coordinator/internal/controller/api/auth.go

```diff
@@ -58,8 +59,19 @@ func (a *AuthController) PayloadFunc(data interface{}) jwt.MapClaims {
 		return jwt.MapClaims{}
 	}
 
+	publicKey := v.PublicKey
+	if publicKey == "" {
+		var err error
+		publicKey, err = v.RecoverPublicKeyFromSignature()
+		if err != nil {
+			// do not handle error here since already called v.Verify() beforehands so there should be no error
+			// add log just in case some error happens
+			log.Error("failed to recover public key from signature", "error", err.Error())
```

Comment:
I think here need return error

Reply:
this method is part of jwt components and no error can be returned in response signature...